docs/src/references/mqtt-api.md

```diff
@@ -0,0 +1,836 @@
+
+### Examples: With default device/service topic semantics
+
+#### Command to main device
+
+Command to fetch the software list from the main device:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/software_list/123' '{
+  "status": "init"
+}'
+```
+
+The `status` value will transition from `init` to the final `successful` or `failed` via many intermediate states
+such as `validating`, `downloading`, `executing` etc.
+
+#### Command to child device
+
+Command to update the firmware of a child device:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01///cmd/firmware_update/123' '{
+  "status": "init",
+  "attempt": 1,
+  "name": "OpenWRT",
+  "version": "22.03",
+  "url":"http://127.0.0.1:8000/tedge/file-transfer/tedge-child/firmware_update/93d50a297a8c235",
+  "sha256":"c036cbb7553a909f8b8877d4461924307f27ecb66cff928eeeafd569c3887e29"
+}'
+```
+
+#### Command to a service
+
+Command to update the configuration of a service:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/service/collectd/cmd/config_update/123' '{
+  "status": "init",
+  "type": "collectd"
+  "path": "/etc/collectd/collectd.conf",
+  "url": "http://127.0.0.1:8000/tedge/file-transfer/collectd/config_update/collectd",
+}'
+```
+
+### Commands metadata
+
+For commands as well, additional command specific metadata can be registered as retained messages on the `<cmd_type>` topic.
+
+For example, the supported configuration list of the main device can be declared as follows:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main///cmd/config_snapshot '{
+  "description": "Upload a configuration from the device",
+  "supportedTypes": ["mosquitto", "tedge", "collectd"]
+}'
+```
+
+*TODO*: Unified config list for both `config_snapshot` and `config_update` commands or support different lists for both?
+
+## Health check
+
+Services can publish their health status as follows:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main/service/tedge-agent/cmd/health_check/ '{
+  "pid": 1234,
+  "status": "successful",
+  "time": 1674739912
+}'
+```
+
+Where the command id in the topic can be skipped with a trailing `/`.
+
+A service can also be queried for its health status by sending a command to it with `init` status as follows:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main/service/tedge-agent/cmd/health_check/ '{
+  "status": "init"
+}'
+```
+
+Services are also expected to react to device-wide health check commands as well:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main///cmd/health_check/ '{
+  "status": "init"
+}'
+```
+
+On receipt of the above command, all services on that device must respond with their health status.
+
+The services are also expected to register a `failed` status last-will message with the broker
+as follows so that the broker updates the health status when that service crashes:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main/service/tedge-agent/cmd/health_check/ '{
+  "status": "failed"
+}'
+```
+
+## Backward compatibility for legacy tedge topics
```

Comment:
I removed the section on the command backward compatibility as I don't think we have enough information as of yet (due to the other ongoing work, https://github.com/thin-edge/thin-edge.io/pull/2071).

Are you ok if we add the compatibility details in later on?

Though generally I am more included to include the "legacy" and new api in all mqtt code blocks throughout the documentation, so that no-one has to keep referring back to this page...e.g. the commands could be automatically transformed using this feature: https://github.com/thin-edge/tedge-docs/pull/36